### ISSUE TITLE
Error while loading test-dev dataset

### ISSUE THREAD

**Issue by omkar13** [fperazzi/davis-2017#10]
The BaseLoader class checks for consistency in the number of sequences and number of annotations. Hence the code breaks. I receive the following error:

Exception: Incorrect frames number for sequence 'aerobatics': found 1, expected 71.

Traceback (most recent call last):
  File "/home/omkar/pycharm-community-2017.3.4/helpers/pydev/pydev_run_in_console.py", line 53, in run_file
    pydev_imports.execfile(file, globals, locals)  # execute the script
  File "/home/omkar/Documents/Omkar/PycharmProjects/Masktrack1/pytorch-deeplab-resnet-master/train_online_multiple_objs&gpus.py", line 234, in <module>
    davis17loader = db17.DAVISLoader(year=cfg.YEAR, phase=cfg.PHASE)
  File "/home/omkar/Documents/Omkar/PycharmProjects/Masktrack1/pytorch-deeplab-resnet-master/dataloaders/davis17.py", line 81, in __init__
    self.annotations = [Annotation(s.name,self._single_object) for s in self._db_sequences]
  File "/home/omkar/Documents/Omkar/PycharmProjects/Masktrack1/pytorch-deeplab-resnet-master/dataloaders/base.py", line 159, in __init__
    osp.join(cfg.PATH.ANNOTATIONS,name),single_object,regex)
  File "/home/omkar/Documents/Omkar/PycharmProjects/Masktrack1/pytorch-deeplab-resnet-master/dataloaders/base.py", line 113, in __init__
    functools.partial(_load_annotation,single_object=single_object))
  File "/home/omkar/Documents/Omkar/PycharmProjects/Masktrack1/pytorch-deeplab-resnet-master/dataloaders/base.py", line 78, in __init__
    self.name,len(self),cfg.SEQUENCES[self.name].num_frames))
Exception: Incorrect frames number for sequence 'aerobatics': found 1, expected 71.

**Comment by fperazzi** [comment]
Is this variable in the config file pointing to the right folder?

# Data folder
__C.PATH.DATA = osp.abspath(osp.join(__C.PATH.ROOT, 'data/DAVIS'))

**Comment by omkar13** [comment]
Yes. But the number of images and number of annotations will not be equal on test data. The test data has only 1 annotation (for the first frame) per sequence. The consistency check should be removed for test data.

**Comment by fperazzi** [comment]
Got it and possibly fixed. Please checkout the latest revision and let me know if that works.

**Comment by shuangjiexu** [comment]
Still get the same error when runing:
`db = DAVISLoader(year="2017", phase=phase.TESTDEV)`.

In my opinion the error is caused in the line 75 of python/lib/davis/dataset/base.py:
`if cfg.PHASE != phase.TESTDEV and len(self) != cfg.SEQUENCES[self.name].num_frames:`
PHASE may use function argument instead of value from config.py directly?
